.github/docker/Dockerfile.bazel

```diff
@@ -1,3 +1,3 @@
-FROM gcr.io/bazel-public/bazel:7.1.0
+FROM gcr.io/bazel-public/bazel:7.0.2
```

Comment:
Something the new bazel is doing is breaking protoc builds. I tested some locally and couldn't find a solution so reverting this back for now.

Build failures look like:
```
$ bazel build '//plugins:protoc-gen-cpp.stripped'
ERROR: Failed to initialize sandbox: /home/build/.cache/bazel/_bazel_build/433461e849f916304f45cf88ae617024/sandbox/_moved_trash_dir -> /home/build/.cache/bazel/_bazel_build/433461e849f916304f45cf88ae617024/sandbox/stale-trash-0 (Invalid cross-device link)
INFO: Elapsed time: 0.198s
INFO: 0 processes.
ERROR: Build did NOT complete successfully
```